engines/wintermute/base/gfx/opengl/base_surface_opengl3d.cpp

```diff
@@ -167,36 +242,34 @@ bool BaseSurfaceOpenGL3D::comparePixel(int x, int y, byte r, byte g, byte b, int
 }
 
 bool BaseSurfaceOpenGL3D::startPixelOp() {
-	glBindTexture(GL_TEXTURE_2D, tex->getTextureName());
+	_imageData = new uint8[4 * _texWidth * _texHeight];
+	glBindTexture(GL_TEXTURE_2D, _tex);
+	glGetTexImage(GL_TEXTURE_2D, 0, GL_RGBA, GL_UNSIGNED_BYTE, _imageData);
 	return true;
 }
 
 bool BaseSurfaceOpenGL3D::endPixelOp() {
 	glBindTexture(GL_TEXTURE_2D, 0);
+	delete[] _imageData;
+	_imageData = nullptr;
 	return true;
 }
 
 bool BaseSurfaceOpenGL3D::isTransparentAtLite(int x, int y) {
-	if (x < 0 || y < 0 || static_cast<uint>(x) >= tex->getWidth() || static_cast<uint>(y) >= tex->getHeight()) {
+	if (x < 0 || y < 0 || x >= _width || y >= _height) {
 		return false;
 	}
 
-	if (!pixelOpReady) {
+	if (_imageData == nullptr) {
 		return false;
 	}
 
-	uint8 *image_data = nullptr;
-
-	// assume 32 bit rgba for now
-	glGetTexImage(GL_TEXTURE_2D, 0, GL_RGBA, GL_UNSIGNED_BYTE, image_data);
-
-	uint32 pixel = *reinterpret_cast<uint32 *>(image_data + y * tex->getWidth() * 4 + x * 4);
-	pixel &= 0x000000FF;
-	return pixel == 0;
+	uint8 alpha = _imageData[y * _texWidth * 4 + x * 4 + 3];
```

Comment:
This assumes the format has a byte order that puts alpha in that position. It would be preferable to query that, as it may not match on Big-Endian systems.

Reply:
If I am not mistaken, than GL_RGBA + UNSIGNED_BYTE means that the alpha component will have offset three here, at least if the texture was uploaded correctly. On the other hand, there are certainly more places where endianess might become an issue. I will have to test things on a big endian platform at some point, I guess.

Reply:
add some comment here about potential endian issue